src/celeritas/user/StepInterface.hh

```diff
@@ -39,8 +39,8 @@ class StepInterface
     using MapVolumeDetector = std::map<VolumeId, DetectorId>;
     //@}
 
-    //! Filtering to apply to the gathered data for this step.
-    struct Filters
+    //! Detector scoring to apply to the gathered data for this step.
+    struct ScoringVolumes
```

Comment:
Yeah, you have point. Man, naming things is hard sometimes...
Ok, one way that we can try to avoid mixing up multiple `filtering` definitions, could be by organizing them by "reference frame" and some usual nomenclature in the field. So here is a possibility:
- Filtering applies to mc truth (the "particle reference frame"): so we filter what particle data we want.
- Scoring and triggering applies to detectors (the "volume reference frame"): we define the scoring volumes/regions as the volumes/regions that are either sensitive or of interest of the user, and we filter what we want (based on energy deposition thresholds, for example) by stating that such conditions are our _triggering_ conditions.

What do you think?

Reply:
I don't get how the two filtering definitions and uses are in conflict, and why they can't have `Filters` classes of their own? Each of the filter definitions is "if enabled, ignore anything that doesn't match these criteria". I don't think we want to use `trigger` because that's tied into digitization and higher-level detector processing.

Reply:
I agree that there is no conflict and we could just use `Filters`. But the `StepInterface` is providing a general interface, being both a `HitManager` or a `SteppingAction`-type of class, such as `RootStepWriter`, that has it's own filter. That means that we could technically have `Filters` for both particle and detector data in `StepInterface` at some point. So all I'm saying is that we could be a bit more explicit in naming that, so in a derived class it's easier to know that when we call `filters()` we actually mean detector filtering.

Reply:
Oh duh, that's what I'm missing. I totally forgot this inherits from the `StepInterface` so there's a name conflict. Since the `RootStepWriter` filter just prevents steps from being written to disk (as opposed to collected in memory), what about restoring the original `filters` for the `StepInterface` classes and just renaming the RSW stuff to  `WriteFilter`?

Reply:
Good point about the difference between the two. Sounds good, I like `WriteFilter`.